tests/test_configurators/test_configurators.py

```diff
@@ -0,0 +1,146 @@
+        'var2': keras.random.normal(test_shape),
+        'var3': keras.random.normal(test_shape),
+        'summary_outputs': keras.random.normal(test_shape),
+    }
+
+
+@pytest.fixture()
+def random_data_no_output(test_shape):
+    return {
+        'var1': keras.random.normal(test_shape),
+        'var2': keras.random.normal(test_shape),
+        'var3': keras.random.normal(test_shape),
+    }
+
+
+@pytest.fixture()
+def test_params():
+    return {
+        'inference_variables': ["var1"],
+        'inference_conditions': ["var2", "var3", "summary_outputs"],
+        'summary_variables': ["var1"],
+        'summary_conditions': ["var2"]
+    }
+
+
+@pytest.fixture()
+def test_params_no_output():
+    return {
+        'inference_variables': ["var1"],
+        'inference_conditions': ["var2", "var3"],
+        'summary_variables': ["var1"],
+        'summary_conditions': ["var2"]
+    }
+
+
+@pytest.fixture()
+def configurator(test_params):
+    return Configurator(
+        inference_variables=test_params['inference_variables'],
+        inference_conditions=test_params['inference_conditions'],
+        summary_variables=test_params['summary_variables'],
+        summary_conditions=test_params['summary_conditions']
+    )
+
+
+@pytest.fixture()
+def configurator_no_output(test_params_no_output):
+    return Configurator(
+        inference_variables=test_params_no_output['inference_variables'],
+        inference_conditions=test_params_no_output['inference_conditions'],
+        summary_variables=test_params_no_output['summary_variables'],
+        summary_conditions=test_params_no_output['summary_conditions']
+    )
+
+
+@pytest.fixture()
+def configurator_sparse(test_params):
+    return Configurator(
+        inference_variables=test_params['inference_variables'],
+    )
+
+
+# Test for correct construction of Configurator with all args
+def test_configurator_init(test_params, configurator: Configurator):
+    config = configurator
+    assert config.inference_variables == test_params['inference_variables']
+    assert config.inference_conditions == test_params['inference_conditions']
+    assert config.summary_variables == test_params['summary_variables']
+    assert config.summary_conditions == test_params['summary_conditions']
+
+
+# Test for correct construction of Configurator with only inference_vars
+def test_sparse_configurator_init(test_params, configurator_sparse: Configurator):
+    config = configurator_sparse
+    assert config.inference_variables == test_params['inference_variables']
+    assert config.inference_conditions == []
+    assert config.summary_variables == []
+    assert config.summary_conditions == []
+
+
+# Test successful configure_inference_variables()
+def test_inference_vars_filter(random_data, configurator: Configurator, test_shape):
+    config = configurator
+    filtered_data = config.configure_inference_variables(random_data)
+    assert filtered_data.shape == test_shape
```

Comment:
This test only works if there is only a single inference_variable, i.e. no concatenation actually happens

Reply:
Refactored test to account for multiple tensors (see b2922ca)